birdhouse/components/stac/config/magpie/stac_hooks.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+"""
+These hooks will be running within Twitcher, using MagpieAdapter context, applied for Stac requests.
+
+The code below can make use of any package that is installed by Magpie/Twitcher.
+
+.. seealso::
+    Documentation about Magpie/Twitcher request/response hooks is available here:
+    https://pavics-magpie.readthedocs.io/en/latest/configuration.html#service-hooks
+"""
+
+import json
+import logging
+import re
+from typing import TYPE_CHECKING
+
+from magpie.api.management.resource import resource_utils as ru
+from magpie.api.requests import get_service_matchdict_checked
+from magpie.models import Route, Service
+from magpie.utils import get_logger
+from magpie.db import get_session_from_other
+
+if TYPE_CHECKING:
+    from pyramid.request import Request
+    from pyramid.response import Response
+
+LOGGER = get_logger("magpie.stac",level=logging.DEBUG)
+
+def create_collection_resource(response):
+    # type: (Response) -> Response
+    """
+    Create the stac collection resource
+    """
+    request = response.request
+    body = request.json
+    collection_id = body['id']
+    display_name = extract_display_name(body['links'])
+
+    # note: matchdict reference of Twitcher owsproxy view is used, just so happens to be same name as Magpie
+    service = get_service_matchdict_checked(request)
+    try:
+        # Getting a new session from the request
+        session = get_session_from_other(request.db)
+        children = ru.get_resource_children(service, db_session=session, limit_depth=2)
+        collection_res_create = True
+        # find the nested resource id matching: "stac/stac"
+        for childs in children.values():
+            if childs["node"].resource_name == "stac":
+                stac_res_id = childs["node"].resource_id
+                for stac_child in childs["children"].values():
+                    # find the nested resource id matching: "stac/stac/collections"
+                    if stac_child["node"].resource_name == "collections":
+                        collection_res_id = stac_child["node"].resource_id
+                        collection_res_create = False
+                        break
+
+        # create resource /stac/stac/collections if does not exist
+        if collection_res_create:
+            collection_res = ru.create_resource("collections", None, Route.resource_type_name, stac_res_id, db_session=session)
+            collection_res_id = collection_res.json["resource"]["resource_id"]
+
+        # create resource /stac/stac/collections/<collection_id>
+        # In try catch since Magpie resource are persistent but stac-db could be wiped
+        try:
+            collection = ru.create_resource(collection_id, display_name, Route.resource_type_name, collection_res_id, db_session=session)
+        except Exception as exc:
+            LOGGER.warning("Failed creation of the collection %s %s", collection_id, str(exc), exc_info=exc)
+
+        session.commit()   
```

Comment:
Session commit should probably be avoided here if the creation failed. 
Calling some abort or rollback method?
And then return the response directly.